### PR TITLE
feat: parse + sort + correlate pipeline glue (W2 #13)

### PR DESCRIPTION
**Pull request by adrian-mason**
## Summary
- Add `src/pipeline.rs` — **parse + file-level sort + correlate glue** from `final-design.md §3.2`
- Chains `WperfReader::read_all_events()` → full-order sort over `WperfEvent::Ord` → `correlate_events()` into a single `build_wait_for_graph()` function
- Also fixes idle thread (tid 0) as waker bug in correlate module (per Gemini review on PR #92)

## Authoritative Inputs
- `final-design.md §3.2` (parse + sort + correlate pipeline)
- `PR #92 / src/correlate.rs` contract (sorted input, last-wake-wins)

## Deviations
- None

## Dependency Checklist
- [x] No new/modified dependencies

## Review Checklist
- [x] **Design conformance**: implementation matches the ADRs/specs listed in Authoritative Inputs — Critic PASS, Challenger PASS
- [x] **Deviations declared**: any spec drift is explicitly listed above with a doc update reference — Critic PASS, Challenger PASS
- [x] **Code correctness**: logic, error handling, edge cases — Critic PASS, Challenger PASS
- [x] **Tests**: new/changed code has adequate test coverage — Critic PASS, Challenger PASS
- [ ] **CI green**: all required checks pass

## Design Decisions
- **Sort**: `events.sort()` using `WperfEvent`'s derived `Ord` for deterministic full-order sorting. Matches `ReorderBuf`'s `BinaryHeap<Reverse<WperfEvent>>` live-path semantics. Do NOT downgrade to `sort_by_key(timestamp_ns)` — see Critic/Challenger plan review.
- **Empty trace**: returns `Ok((empty graph, zero stats))`, not a hard error. CLI layer decides UX.
- **Idle waker fix**: wakeup events with source=0 (kernel idle) now skipped, preventing edges to `ThreadId(0)`.

## Test Plan
- 7 pipeline tests: empty trace, single event, simple chain, unsorted events sorted, unmatched wakeup stats, error display, error source
- 1 new correlate test: `wakeup_from_idle_ignored`
- `cargo clippy --all-targets -- -D warnings` clean
- `cargo fmt -- --check` clean
- Full test suite: 195/195 pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/pipeline.rs [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+/// 2. **Sort**: full-order sort over `WperfEvent::Ord` for deterministic
+///    ordering (see module docs)
+/// 3. **Correlate**: [`correlate_events`] produces edges and stats
+///
+/// Empty traces return `Ok` with an empty graph and zero stats.
+pub fn build_wait_for_graph<R: Read + Seek>(
+    reader: &mut WperfReader<R>,
+) -> Result<(WaitForGraph, PipelineStats), PipelineError> {
+    // Step 1: Parse
+    let mut events = reader.read_all_events()?;
```
[the remark this answers is not in the document]
Comment (adrian-mason) [reply src/pipeline.rs:83]: @gemini-code-assist Valid optimization idea — agreed that pre-allocating the event Vec would reduce reallocs on large traces.

However, the suggested patch applies format-layer constants (`HEADER_SIZE`, `TLV_HEADER + EVENT_SIZE = 45`) at the pipeline layer, which leaks `.wperf` TLV layout knowledge out of the reader module. It also uses `header.data_section_end_offset` directly, bypassing the crash-recovery clamp (`data_end = header.data_section_end_offset.min(file_len)`) that `WperfReader::open()` applies internally — on a stale/corrupt header this could over-allocate.

If we take this optimization, the right place is inside `WperfReader::read_all_events()` using the already-clamped `data_end` and the reader's own format constants. Tracked as a non-blocking perf follow-up.
